plugins/modules/pulp_ansible_distribution.py

```diff
@@ -0,0 +1,145 @@
+---
+module: pulp_ansible_distribution
+short_description: Manage ansible distributions of a pulp server
+description:
+  - "This performes CRUD operations on ansible distributions in a pulp server."
+options:
+  name:
+    description:
+      - Name of the distribution to query or manipulate
+    type: str
+    required: false
+  base_path:
+    description:
+      - Base path to distribute a publication
+    type: str
+    required: false
+  repository:
+    description:
+      - Name of the repository to be served
+    type: str
+    required: false
+  version:
+    description:
+      - Version number of the repository to be served
+      - If not specified, the distribution will always serve the latest version.
+    type: int
+    required: false
+  content_guard:
+    description:
+      - Name of the content guard for the served content
+    type: str
+    required: false
+extends_documentation_fragment:
+  - mdellweg.squeezer.pulp
+  - mdellweg.squeezer.pulp.entity_state
+author:
+  - Matthias Dellweg (@mdellweg)
+'''
+
+EXAMPLES = r'''
+- name: Read list of ansible distributions from pulp api server
+  pulp_ansible_distribution:
+    api_url: localhost:24817
+    username: admin
+    password: password
+  register: distribution_status
+- name: Report pulp ansible distributions
+  debug:
+    var: distribution_status
+
+- name: Create an ansible distribution
+  pulp_ansible_distribution:
+    api_url: localhost:24817
+    username: admin
+    password: password
+    name: new_ansible_distribution
+    base_path: new/ansible/dist
+    repository: my_repository
+    state: present
+
+- name: Delete an ansible distribution
+  pulp_ansible_distribution:
+    api_url: localhost:24817
+    username: admin
+    password: password
+    name: new_ansible_distribution
+    state: absent
+'''
+
+RETURN = r'''
+  distributions:
+    description: List of ansible distributions
+    type: list
+    returned: when no name is given
+  distribution:
+    description: Ansible distribution details
+    type: dict
+    returned: when name is given
+'''
+
+
+from ansible_collections.mdellweg.squeezer.plugins.module_utils.pulp_helper import PulpEntityAnsibleModule
+from ansible_collections.mdellweg.squeezer.plugins.module_utils.pulp_ansible_helper import PulpAnsibleDistribution, PulpAnsibleRepository
+
+
+def main():
+    with PulpEntityAnsibleModule(
+        argument_spec=dict(
+            name=dict(),
+            base_path=dict(),
+            repository=dict(),
+            version=dict(type='int'),
+            content_guard=dict(),
+        ),
+        required_if=[
+            ('state', 'present', ['name', 'base_path']),
+            ('state', 'absent', ['name']),
+        ],
+    ) as module:
+
+        repository_name = module.params['repository']
+        version = module.params['version']
+
+        if module.params['content_guard']:
+            module.fail_json(msg="Content guard features are not yet supportet in this module.")
```

Comment:
Should this perhaps be stated in `DOCUMENTATION` under `content_guard`?

Reply:
good point